tensorflow_datasets/image/malaria_bbbc.py

```diff
@@ -0,0 +1,98 @@
+# coding=utf-8
+# Copyright 2019 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import fnmatch
+import os
+
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+_URL = "http://people.duke.edu/~kk349/bbbc.zip"
+
+_CITATION = """\
+ @misc{ljosa_sokolnicki_carpenter_2012,
+  title={Annotated high-throughput microscopy image sets for validation},
+  author={Ljosa, Vebjorn and Sokolnicki, Katherine L and Carpenter, Anne E},
+  url={https://www.nature.com/articles/nmeth.2083},
+  journal={Nature News},
+  publisher={Nature Publishing Group},
+  year={2012},
+  month={Jun}
+}
+"""
+
+_DESCRIPTION = """The original data set contains 1364 images and
+consists of two classes of uninfected cells(red blood cells and leukocytes) and
+four classes of infected cells(gametocytes, rings, schizonts, and trophozoites).
+The Malaria dataset contains a total of 74,238 cell images and we have removed 441 cell images
+because annotators marked the cells as difficult as it was not clearly in one of the cell classes.
+Fianlly malaria_bbbc data set contains 73,797 cells.
+This data set is extremly skewed toward uninfected cases, so we did not divide original data to train and test sets
+so that the user can try in their own way to deal with this problem.
+"""
+
+_NAMES = ["gametocyte", "leukocyte", "red blood cell", "ring", "schizont", "trophozoite"]
+
+_IMAGE_SHAPE = (None, None, 3)
+
+
+class MalariaBbbc(tfds.core.GeneratorBasedBuilder):
+  """malaria_bbbc dataset."""
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=_IMAGE_SHAPE),
+            "label": tfds.features.ClassLabel(names=_NAMES),
+        }),
+        supervised_keys=("image", "label"),
+        urls=[_URL],
```

Comment:
this url should be dataset home page link not download link.

Reply:
@us The dataset has the following copyright access. Free to redistribute.
https://creativecommons.org/licenses/by-nc-sa/3.0/